### ISSUE TITLE
Help: How do I make `history.pushState()` accept an `URL` instance?

### ISSUE THREAD

**Issue by Kocal** [microsoft/TypeScript-DOM-lib-generator#670]
Hi,

I opened this PR (https://github.com/Microsoft/TypeScript/pull/30512) to make the following code compilable with TypeScript:
```js
const url = new URL(location.href)
url.search = '?foo=bar'
history.pushState({}, 'Title', url)
```

The following code is working, (you can try it in your console, it will append `?foo=bar` to the URL), but we can't compile it with TypeScript:
![Sélection_026](https://user-images.githubusercontent.com/2103975/54716703-9de7de00-4b56-11e9-8898-2a5d8a4876b9.jpg)

The current signature is
```ts
replaceState(data: any, title: string, url?: string | null): void;
```

and I want to update it to:
```ts
replaceState(data: any, title: string, url?: string | URL | null): void;
```

So as @typescript-bot said (https://github.com/Microsoft/TypeScript/pull/30512#issuecomment-475014366), I should open a PR here but I don't know what to update.

After some research, `replaceState` appears three times in the following files:
  - `dom.generated.d.ts`, but I should not update this file manually because it's auto-generated
  - `browser.webidl.preprocessed.json`, I should not update it either
  - `HTML - Session history and navigation.widl`, but the signature is `void replaceState(any data, DOMString title, optional USVString? url = null)` and I don't understand how `optional USVString? url = null` is transformed to `url?: string | null` :thinking: 

Even in the `overridingTypes.json` file, I don't see any `pushState()` / `replaceState()` :confused: 

So what should I do to make `replaceState()` and `pushState()` accepts an `URL` instance as the 3rd parameter?

Thanks! :)

**Comment by saschanaz** [comment]
@ᅟsandersn We keep facing this... Maybe we could add a rule: *if the argument name is `url` and its type is `string`, then allow `URL` too.* Thoughts?

**Comment by saschanaz** [comment]
Implemented by #1034

@github-actions close
